### PR TITLE
DOC: fix SA04 errors flagged by validate_docstrings.py and add SA04 t…

### LINKED ISSUE #25337: Fix validation error type `SA04` and check in CI
#### Problem description
In order to have a continuous check by the CI on validation error `SA04` (Missing description for See Also reference), fixing them in the code base enables the addition to the CI for automated testing in the future.

Please see [overview of the errors](https://gist.github.com/198cfa552c727d71bd7360fda66fdbb4).

Todo:

* [ ]     get rid of the errors in the code base
* [ ]     update the `code_check.sh` script to take into account the `SA04` type of errors

### PR DESCRIPTION
**Pull request by dataxerik**
- [x] closes #25337
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [NA] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Many of the SA04 reported in #25337 are already fixed, but there were two outstanding SA04 errors reported by valdiate_docstrings.py on main. So, I fixed those and added 'SA04' to the code_checks.py as per #25337

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4371,8 +4371,8 @@ def reindex(
 
         See Also
         --------
-        Series.reindex
-        DataFrame.reindex
+        Series.reindex : Return only specified index labels of Series.
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:4374]: I think these descriptions should be be same except mentioning which object they apply to.

You can copy the first sentence of their docstring:

```
Conform Series/DataFame to new index with optional filling logic.
```

Comment (dataxerik) [reply pandas/core/indexes/base.py]: I originally took the text from another See Also section. This sounds like a good change, and I've made the update. Thank you for the feedback!

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -4371,8 +4371,9 @@ def reindex(
 
         See Also
         --------
-        Series.reindex
-        DataFrame.reindex
+        Series.reindex : Conform Series to new index with optional filling logic.
+        DataFrame.reindex : Conform Series/DataFrame to new index with optional filling
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:4375]: `Series/DataFrame` -> `Series`

Comment (dataxerik) [reply pandas/core/indexes/base.py]: Updated. Thanks again for the feedback

Comment (dataxerik) [reply pandas/core/indexes/base.py]: Updated. Thanks again for the feedback

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @dataxerik

**Pull request merged**